### DISCUSSION TITLE
Identify can notify about listen address changes, but not external address changes, why?

### DISCUSSION THREAD

**Discussion by nazar-pc** [libp2p/rust-libp2p#4867] [Q&A]
I'm wondering because starting with Autonat and automatic Kademlia mode node will start as a client and the other side will never know that the client has become a server.

With automatic identify push (optionally) it would be possible to update Kademlia on the other end to contain updated addresses.

For now I'll be calling `.push()` manually, but another boolean option would probably be helpful.

**Comment by thomaseizinger** [comment]
IMO this is a bit of a misnomer. The [`listenAddr`](https://github.com/libp2p/specs/tree/master/identify#listenaddrs) field in fact contains external addresses. In fact, I think we [should remove](https://github.com/libp2p/rust-libp2p/issues/4010) listen addresses from that field entirely because they are likely useless to the remote.

**Reply by nazar-pc** [reply]
I actually did see that the naming is confusing (and was planning to report that, but forgot).

What I meant here is that there is `push_listen_addr_updates`, but I'd like to have `push_external_addr_updates` because I agree listen address updates are not really that helpful, I wouldn't want to enable it.

**Reply by thomaseizinger** [reply]
Ah I see. I agree that this option makes sense. I think it is also something we can enable by default.
